app/assets/javascripts/discourse/app/components/modal/sidebar-section-form.js

```diff
@@ -232,22 +232,22 @@ class SectionLink {
   }
 }
 
-export default Controller.extend(ModalFunctionality, {
-  dialog: service(),
-  router: service(),
+export default class SidebarSectionForm extends Component {
+  @service dialog;
+  @service router;
 
-  onShow() {
-    this.setProperties({
-      flashText: null,
-      flashClass: null,
-    });
+  @tracked flash;
+  @tracked flashType;
+
+  init() {
     this.nextObjectId = 0;
     this.model = this.initModel();
-  },
+    super.init(...arguments);
+  }
 
-  onClose() {
+  willDestroy() {
     this.model = null;
-  },
+  }
```

Comment:
Now that this is a short-lived component, there's no need to clean up state manually. All internal state will be blown away when the component is destroyed.